packages/web/src/session/cookie-session.ts

```diff
@@ -15,23 +15,39 @@
  * limitations under the License.
  *
  */
-import { isIframe } from './utils'
+import { isIframe } from '../utils'
 import { SessionState } from './types'
-
-export const COOKIE_NAME = '_splunk_rum_sid'
-
-const CookieSession = 4 * 60 * 60 * 1000 // 4 hours
-const InactivityTimeoutSeconds = 15 * 60
+import { SESSION_DURATION_SECONDS, SESSION_STORAGE_KEY } from './constants'
+import { isSessionDurationExceeded, isSessionInactivityTimeoutReached, isSessionState } from './utils'
+import { throttle } from '../utils/throttle'
 
 export const cookieStore = {
-	set: (value: string): void => {
+	cachedValue: null,
+	set: (value: string) => {
+		cookieStore.cachedValue = value
+		cookieStore._set(value)
+	},
+
+	_set: throttle((value: string) => {
```

Comment:
Writing to cookies/localstorage is throttled. We write directly on load, when session is created and when page gets hidden.